enginetest/join_op_tests.go

```diff
@@ -1357,4 +1358,192 @@ SELECT SUM(x) FROM xy WHERE x IN (
 			},
 		},
 	},
+	{
+		name: "primary key range join",
```

Comment:
a couple edge cases customers might hit:
- left joins
- join relations being non-standard types (table alias, subquery alias, CTE, recursive CTE, table function, tables with mixed upper/lower casing)
- one of the tables being empty
- one of the tables having 1 row
- the tables matching cartesian product of both tables
- tables matching no rows

Reply:
other misc maybe lower priority:
- new join op plays nicely with DISTINCT, projections, subquery expressions
- new join op can be disabled with hints

Reply:
Added.